File: OpenMP/solutions/omp_vectadd.f
      program vectoradd
             
        use omp_lib
        implicit none

c       Initialize variables
        integer :: size = 1000000
        integer, allocatable :: a(:)
        integer, allocatable :: b(:)
        integer, allocatable :: c(:)
        double precision :: start, end
        integer :: i
        double precision :: val

        allocate(a(size))
        allocate(b(size))
        allocate(c(size))

c       Time the parallel loop

        start = omp_get_wtime()

c       Fill a & b with random numbers beween 0 and 100
c       IMPORTANT NOTES:
c        * rand() is only valid for GNU Fortran compilers
c        * random_number is only valid for Fortran 95 or later
!$OMP PARALLEL DO
        do i=1,size
          call random_number(val)
          a(i) = nint(val*100)
          call random_number(val)
          b(i) = nint(val*100)
          c(i) = a(i) + b(i)
        enddo
!$OMP END PARALLEL DO

        end = omp_get_wtime() - start;

        deallocate(a)
        deallocate(b)
        deallocate(c)

        print *,"Summed vectors a and b in ",end," seconds"


      end program

File: OpenMP/exercises/omp_vectadd.c
#include <omp.h>
#include <stdio.h>
#include <stdlib.h>

int main (int argc, char *argv[])
{

  // Initialize arrays 
  int size=1000000;
  int *a = (int *)malloc(size*sizeof(int));
  int *b = (int *)malloc(size*sizeof(int));
  int *c = (int *)malloc(size*sizeof(int));

  // Time the parallel for-loop
  double start,end;
  start = omp_get_wtime();

  // Fill with a & b with random numbers between 0 and 100
  int i;
  for(i=0; i<size; i++)
  {
    a[i]=rand() % 100;
    b[i]=rand() % 100;
    c[i]=a[i]+b[i];
  }

  end = omp_get_wtime();

  free(a);
  free(b);
  free(c);

  printf("Summed vectors a and b in %g seconds\n", end-start);

}
